### PR TITLE
fix uv switch

### PR DESCRIPTION
**Pull request by samuelcolvin**
correct use of `uv`.

### CODE REVIEW

File: tests/test_pickle.py [hunk 1]
```diff
@@ -3,14 +3,20 @@
...
 from pydantic import BaseModel, PositiveFloat, ValidationError
 from pydantic._internal._model_construction import _PydanticWeakRef
 from pydantic.config import ConfigDict
 
+try:
+    import cloudpickle
+except ImportError:
+    cloudpickle = None
+
+pytestmark = pytest.mark.skipif(cloudpickle is None, reason='cloudpickle is not installed')
```
Comment (sydney-runkle) [inline tests/test_pickle.py:18]: Not every test here requires `cloudpickle`, right?

**Review by sydney-runkle — APPROVED** [review]
LGTM, thanks! Just one follow up question.

**Comment by sydney-runkle** [comment]
Re https://github.com/pydantic/pydantic/pull/10815/commits/eedf5567bb19c9b1758886148339b4939e4cc649:

I don't think this should be added to `dev`, as we don't have `email-validator` there. They're both extras...

**Comment by sydney-runkle** [comment]
Also, we should investigate the perf bottlenecks here - very odd...

**Comment by davidhewitt** [comment]
Perf regressions are because `pytest-memray` is now being installed; for whatever reason our benchmarks are measuring pytest hook execution time...

image

**Comment by davidhewitt** [comment]
Looks like `@pytest.mark.benchmark` is the problematic pattern, measuring the wrong thing.

**Comment by samuelcolvin** [comment]
@sydney-runkle I think this is ready.

**Comment by sydney-runkle** [comment]
@samuelcolvin,

LGTM, though I don't think you addressed this one: https://github.com/pydantic/pydantic/pull/10815#issuecomment-2468485315

**Comment by samuelcolvin** [comment]
> Re [eedf556](https://github.com/pydantic/pydantic/commit/eedf5567bb19c9b1758886148339b4939e4cc649):
> 
> I don't think `tzdata` should be added to `dev`, as we don't have `email-validator` there. They're both extras... So, we should probably add conditional skips to those tests accordingly, like you've done for the email validator ones.

It has to be included for windows tests, hence the hint.

**Comment by sydney-runkle** [comment]
Alright! Let's merge then.

Thanks for the improvements here 👍, looking much cleaner.

**Pull request merged**